clang/runtime/dpct-rt/include/sparse_utils.hpp

```diff
@@ -499,7 +506,17 @@ class sparse_matrix_desc {
   /// numbering.
   /// \param [in] value An array containing the non-zero elements of the sparse matrix.
   void set_pointers(void *row_ptr, void *col_ind, void *value) {
-    // Assume the new data is different from the old data
+    if (_row_ptr != row_ptr) {
+      throw std::runtime_error("the row_ptr cannot be changed");
+    }
+    if (_col_ind) {
+      throw std::runtime_error("the col_ind pointer can be reset only when the "
+                               "original pointer is NULL");
+    }
+    if (_value) {
+      throw std::runtime_error("the value pointer can be reset only when the "
+                               "original pointer is NULL");
+    }
     _row_ptr = row_ptr;
     _col_ind = col_ind;
     _value = value;
```

Comment:
call set_value() here, and add a comment to note the descriptor will be updated in the set_value